homeassistant/components/blue_current/button.py

```diff
@@ -0,0 +1,101 @@
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from bluecurrent_api.client import Client
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Connector
+from .const import DOMAIN
+from .entity import ChargepointEntity
+
+
+@dataclass(frozen=True)
+class ChargePointButtonEntityDescriptionMixin:
+    """Mixin for the called functions."""
+
+    function: Callable[[Client, str], Coroutine[Any, Any, None]]
+
+
+@dataclass(frozen=True)
+class ChargePointButtonEntityDescription(
+    ButtonEntityDescription, ChargePointButtonEntityDescriptionMixin
+):
+    """Describes a Blue Current button entity."""
+
+
+CHARGE_POINT_BUTTONS = (
+    ChargePointButtonEntityDescription(
+        key="reset",
+        translation_key="reset",
+        function=lambda client, evse_id: client.reset(evse_id),
+        device_class=ButtonDeviceClass.RESTART,
+    ),
+    ChargePointButtonEntityDescription(
+        key="reboot",
+        translation_key="reboot",
+        function=lambda client, evse_id: client.reboot(evse_id),
+        device_class=ButtonDeviceClass.RESTART,
+    ),
+    ChargePointButtonEntityDescription(
+        key="stop_charge_session",
+        translation_key="stop_charge_session",
+        function=lambda client, evse_id: client.stop_session(evse_id),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Blue Current buttons."""
+    connector: Connector = hass.data[DOMAIN][entry.entry_id]
+    button_list: list[ButtonEntity] = [
+        ChargePointButton(
+            connector,
+            button,
+            evse_id,
+        )
+        for evse_id in connector.charge_points
+        for button in CHARGE_POINT_BUTTONS
+    ]
+
+    async_add_entities(button_list)
+
+
+class ChargePointButton(ChargepointEntity, ButtonEntity):
+    """Define a charge point button."""
+
+    def __init__(
+        self,
+        connector: Connector,
+        button: ChargePointButtonEntityDescription,
+        evse_id: str,
+    ) -> None:
+        """Initialize the button."""
+        super().__init__(connector, evse_id)
+
+        self.has_value = True
+        self.function = button.function
+        self.entity_description = button
+        self._attr_unique_id = f"{button.key}_{evse_id}"
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        await self.function(self.connector.client, self.evse_id)
+
+    @callback
+    def update_from_latest_data(self) -> None:
+        """Update the entity from the latest data."""
```

Comment:
I'm not sure why this one is here?

Seems to be caused some dispatcher logic in the base entity. This should instead have used an data update coordinator, so this isn't needed.

Reply:
I have moved the Connector class (in __init__.py) to a Coordinator class. Should i create a separate pr for that? I did add the other requested changes to this pr.